ecl/hql/hqlgram.y

```diff
@@ -6495,6 +6495,14 @@ primexpr1
                         {
                             $$.setExpr(createValue(no_matched, makeBoolType(), $3.getExpr())); //, parser->createUniqueId()));
                         }
+    | MATCHED '(' dataRow ')'
+                        {
+                            $$.setExpr(createValue(no_matched_injoin, makeBoolType(), $3.getExpr()));
```

Comment:
trivial: Should add an extra $1 parameter to the setExpr to record the source code position.